aiida/storage/psql_dos/backend.py

```diff
@@ -453,3 +450,96 @@ def get_info(self, detailed: bool = False) -> dict:
+            incremental and efficient.
+
+        :return:
+            True is successful and False if unsuccessful.
+        """
+        import os
+        import subprocess
+        import tempfile
+
+        from aiida.common import exceptions
+        from aiida.common.exceptions import LockedProfileError
+        from aiida.manage.configuration import get_config
+        from aiida.manage.profile_access import ProfileAccessManager
+
+        cfg = self._profile.storage_config
+        container = Container(str(get_filepath_container(self.profile)))
+
+        # check that the AiiDA profile is not locked and request access for the duration of this backup process
+        # (locked means that possibly a maintenance operation is running that could interfere with the backup)
+        try:
+            ProfileAccessManager(self._profile).request_access()
+        except LockedProfileError:
+            raise backup_utils.BackupError('The profile is locked!')
+
+        # step 1: first run the storage maintenance version that can safely be performed while aiida is running
+        self.maintain(full=False, compress=True)
+
+        # step 2: dump the PostgreSQL database into a temporary directory
+        pg_dump_exe = manager.exes['pg_dump']
+        with tempfile.TemporaryDirectory() as temp_dir_name:
+            psql_temp_loc = pathlib.Path(temp_dir_name) / 'db.psql'
+
+            env = os.environ.copy()
+            env['PGPASSWORD'] = cfg['database_password']
+            cmd = [
+                pg_dump_exe, f'--host={cfg["database_hostname"]}', f'--port={cfg["database_port"]}',
+                f'--dbname={cfg["database_name"]}', f'--username={cfg["database_username"]}', '--no-password',
+                '--format=p', f'--file={str(psql_temp_loc)}'
+            ]
+            try:
+                subprocess.run(cmd, check=True, env=env)
+            except subprocess.CalledProcessError as exc:
+                raise backup_utils.BackupError(f'pg_dump: {exc}')
+
+            if psql_temp_loc.is_file():
+                STORAGE_LOGGER.info(f'Dumped the PostgreSQL database to {str(psql_temp_loc)}')
+            else:
+                raise backup_utils.BackupError(f"'{str(psql_temp_loc)}' was not created.")
+
+            # step 3: transfer the PostgreSQL database file
+            manager.call_rsync(psql_temp_loc, path, link_dest=prev_backup, dest_trailing_slash=True)
+
+        # step 4: back up the disk-objectstore
+        backup_utils.backup_container(
+            manager, container, path / 'container', prev_backup=prev_backup / 'container' if prev_backup else None
+        )
+
+        # step 5: back up aiida config.json file
+        try:
+            config = get_config()
+            manager.call_rsync(pathlib.Path(config.filepath), path)
+        except (exceptions.MissingConfigurationError, exceptions.ConfigurationError):
+            STORAGE_LOGGER.warning('aiida config.json not found!')
```

Comment:
This backs up the entire config, but really we are just backing up a single profile. I wonder if we should backup just the information of this profile. We would have to include the information of the config version if a restore method would ever want to be able to read it.

Reply:
thanks. I just did it like this to be as safe and as simple as possible (e.g. if someone wants to manually replicate this process, it's easy to say, copy also the config.json). The relevant data should be available in the config.json for restoring in the future.

But now that i think about it, I don't think there is a clear/simple indication in the backup folder regarding what profile was backed up. I'm wondering what's the best way to add this. Maybe as you suggest, filter out all other profiles from config.json? or maybe just make a root-level file called profile_<profile_name>? or something else? What do you think?

Reply:
It is relatively straight forward to create a clone of the config file that contains _just_ the backed up profile. It will include the version information of the config file itself, and any options specific to the profile. It should therefore be complete and minimal at the same time to be able to restore from at a later point in time. It should look something like:
```python
import pathlib
import tempfile
from aiida.manage.configuration import get_config, Config
config = get_config()
profile = config.get_profile('profile_name_that_is_being_backed_up')  # Get the profile being backed up
with tempfile.TemporaryDirectory() as tmpdir:
    filepath_config = pathlib.Path(tmpdir) / 'config.json'
    backup_config = Config(filepath_config , {})  # Create empty config at temporary file location
    backup_config.add_profile(profile)  # Add the profile being backed up
    backup_config.store()  # Write the contents to disk
```

I think this would be a good approach. What do you think?

Reply:
thanks, looks good, will add this.